utils/genesis/src/lib.rs

```diff
@@ -114,17 +115,42 @@ where
     info!("Importing chain from snapshot at: {path}");
     // start import
     let stopwatch = time::Instant::now();
-    let (cids, n_records) = if is_remote_file {
+    let import_chain_task = if is_remote_file {
         info!("Downloading file...");
         let url = Url::parse(path)?;
         let reader = get_fetch_progress_from_url(&url).await?;
-        load_and_retrieve_header(sm.blockstore().clone(), reader, skip_load).await?
+        Either::Left(load_and_retrieve_header(
+            sm.blockstore().clone(),
+            reader,
+            skip_load,
+        ))
     } else {
         info!("Reading file...");
         let reader = get_fetch_progress_from_file(&path).await?;
-        load_and_retrieve_header(sm.blockstore().clone(), reader, skip_load).await?
+        Either::Right(load_and_retrieve_header(
+            sm.blockstore().clone(),
+            reader,
+            skip_load,
+        ))
     };
 
+    // Not checking `validate_height` to decide whether to download parameter files
+    // here because it defaults to `None` in config, making parallelization
+    // disabled by default
+    let ensure_params_downloaded_task_handle =
+        tokio::spawn(forest_utils::proofs_api::paramfetch::ensure_params_downloaded());
```

Comment:
I am not convinced that `import_chain` is a good place for spawning the params download task. I mean, as a developer, I wouldn't expect such a method to download lots of relatively large files.

Could we somehow refactor this to have those separated? I don't have anything precise in mind. Perhaps splitting the method into smaller ones and a bit of renaming will do.

Reply:
Done